### PR TITLE
Fixing inconsistency of #157 merged too early

### PR DESCRIPTION
**Pull request by Ph0tonic**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/drone/Drone.kt [hunk 1]
```diff
@@ -102,13 +104,16 @@ object Drone {
...
                     .get(WAIT_TIME, TimeUnit.MILLISECONDS).isConnected
-
         } catch (e: TimeoutException) {
             false
         }
     }
 
     fun isFlying(): Boolean {
-        return instance.mission.isMissionFinished.blockingGet()
+        return try {
```
Comment (AlexisTabin) [inline app/src/main/java/ch/epfl/sdp/drone/Drone.kt:113]: This function is useless (and not working,  but idk why), there is a attribut inAir in Telemetry, but I am changing that in my PR

File: app/src/androidTest/java/ch/epfl/sdp/MapActivityTest.kt [hunk 2]
```diff
@@ -96,43 +99,44 @@ class MapActivityTest {
...
+                .putString(applicationContext().getString(R.string.prefs_drone_altitude), "20")
+                .apply()
+
         // Launch activity
         mActivityRule.launchActivity(intentWithGroupAndOperator)
         mUiDevice.wait(Until.hasObject(By.desc(applicationContext().getString(R.string.map_ready))), MAP_LOADING_TIMEOUT)
         assertThat(mActivityRule.activity.mapView.contentDescription == applicationContext().getString(R.string.map_ready), equalTo(true))
-        mActivityRule.activity.isTest = true
-
-        val expectedLatLng = LatLng(47.397026, 8.543067)
```
Comment (Sporarum) [inline app/src/androidTest/java/ch/epfl/sdp/MapActivityTest.kt:106]: Why did you remove the variable ?
In my opinion, it made the test easier to read

File: app/src/androidTest/java/ch/epfl/sdp/MapActivityTest.kt [hunk 3]
```diff
@@ -96,43 +99,44 @@ class MapActivityTest {
...
+        runOnUiThread {
+            mActivityRule.activity.launchMission()
+        }
 
         val uploadedMission = Drone.currentMissionLiveData.value
 
-            assertThat(uploadedMission, `is`(notNullValue()))
-            assertThat(uploadedMission!!.size, not(equalTo(0)))
-            assertThat(uploadedMission[0].latitudeDeg, closeTo(expectedLatLng.latitude, EPSILON))
-            assertThat(uploadedMission[0].longitudeDeg, closeTo(expectedLatLng.longitude, EPSILON))
```
Comment (Sporarum) [inline app/src/androidTest/java/ch/epfl/sdp/MapActivityTest.kt:133]: Those lines test if the first point of the mission is the closest to the starting point, and should not be removed

Comment (Ph0tonic) [reply app/src/androidTest/java/ch/epfl/sdp/MapActivityTest.kt:133]: This test is way more complexe and should only test if it launch a mission the part which test if the startegy take into account the starting point should be tested inside our strategy tests.

Comment (Ph0tonic) [reply app/src/androidTest/java/ch/epfl/sdp/MapActivityTest.kt:133]: Further more, this specific part doesn't have any sens if we use another strategy.

Comment (AlexisTabin) [reply app/src/androidTest/java/ch/epfl/sdp/MapActivityTest.kt:133]: Okay, but then there should be a test that test whether or not the mission is launched in the right way

Comment (Sporarum) [reply app/src/androidTest/java/ch/epfl/sdp/MapActivityTest.kt:133]: I agree, however the test should then be moved instead of being removed, no ?

Comment (Ph0tonic) [reply app/src/androidTest/java/ch/epfl/sdp/MapActivityTest.kt:133]: @Sporarum yes, I added a todo.

Comment (Sporarum) [reply app/src/androidTest/java/ch/epfl/sdp/MapActivityTest.kt:133]: I don't see it, where is it ?
Shouldn't you then leave this test complete and TODO the cleaning part ?

Comment (Ph0tonic) [reply app/src/androidTest/java/ch/epfl/sdp/MapActivityTest.kt:133]: It's here: #c9b08e2

Comment (Sporarum) [reply app/src/androidTest/java/ch/epfl/sdp/MapActivityTest.kt:133]: Thank you, I still think you should move the TODO and leave this test with everything it's testing
